### DISCUSSION TITLE
단축 평가를 활용해 기본값을 설정할 수 있나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#35] [타입변환과 단축 평가]
- 단축 평가를 활용해 기본값을 설정할 수 있나요?

**Comment by pipisebastian** [comment]
> 단축 평가를 활용해 기본값을 설정할 수 있나요?

논리합 연산자(||)를 통해 기본값을 설정할 수 있습니다. 논리합 연산자는 좌항의 피연산자가 Falsy값일 경우, 우항의 피연산자를 반환하므로, 
우항에 기본값을 넣어준다면, 비어있는 좌항대신 우항을 기본값으로 설정할 수 있습니다.

그러나 논리합 연산자는 Falsy를 기준으로 판단하기에, 좌항이 0, '' 일때도 기본값을 설정해버립니다. 
만약 값이 없을때, 즉 값이 undefined/null일때만 기본값을 설정하고 싶다면, 논리합 연산자 대신 null 병합 연산자를 사용하는 것이 좋습니다.

**Comment by publdaze** [comment]
> - 단축 평가를 활용해 기본값을 설정할 수 있나요?
- 논리합 연산자를 사용해 기본값 설정 가능. falsy 값일 경우 기본값으로 설정하도록 할 수 있음
